client/src/js/auth-helpers.js

```diff
@@ -0,0 +1,103 @@
+import { h, Component, render } from 'preact';
+import Cookies from 'universal-cookie';
+import { route } from 'preact-router';
+import decode from 'jwt-decode';
+import { Menu } from 'preact-material-components';
+import {API_SERVER} from "../../config";
+import axios from "axios/index";
+
+const tokenCookie = 'token';
+
+export const logout = () => {
+	deleteTokenCookie();
+	route('/',true);
+}
+
+export const deleteTokenCookie = () => {
+	const cookies = new Cookies();
+	cookies.remove(tokenCookie);
+}
+
+export const setTokenCookie = (val) => {
+	const cookies = new Cookies();
+	cookies.set(tokenCookie, val, { path: '/' });
+}
+
+export const get_sign_in_out = () => {
+	if(isSignedIn()){
+		return (<a href="" onClick={() => logout()}><Menu.Item>Sign Out</Menu.Item></a>);
+	}else{
+		return (<a href="/signin"><Menu.Item>Sign In</Menu.Item></a>);
+	}
+}
+
+export const isSignedIn = () => {
+	const cookies = new Cookies();
+	const encodedToken = cookies.get('token');
+	if(encodedToken === undefined) return false;
+
+	const token = decode(encodedToken);
```

Comment:
Maybe I'm over thinking this but a little concerned we're decoding the token on the frontend...I feel like we shouldn't be doing so. We have an end point for getting user information from the server once authenticated so this may not be necessary. @motosharpley what do you think?

@seckboy I assume you were doing this to get access to when it expires? This can be handled directly by the cookie options noted earlier.

Reply:
ya the verification / decoding should only be happening on the server side. If the token is expired or invalid we should require the user to authenticate

Reply:
The token should be hashed on the client and then checked on the server. I'll try to explain better tomorrow.